app/src/main/java/com/sanjeevyadavit/magecart/di/AppModule.kt

```diff
@@ -0,0 +1,27 @@
+package com.sanjeevyadavit.magecart.di
+
+import com.sanjeevyadavit.magecart.data.remote.ApiInterface
+import com.sanjeevyadavit.magecart.data.repository.MageCartRepositoryImpl
+import com.sanjeevyadavit.magecart.domain.repository.MageCartRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+class AppModule {
+
+    @Provides
+    @Singleton
+    fun provideApiInterface(): ApiInterface {
+        return RetrofitClient.getInstance()
+    }
```

Comment:
You should also make retrofit injectable and the create the service's instance here and use better naming. The current line 19 is making me understand that i am getting a retrofit instance rather than an instance of ApiInterface.